app/infrastructure/scrapers/lms_scraper.py

```
@@ -19,376 +19,182 @@
+        if not course_list:
+            raise ScraperException(error_type=ErrorType.COURSE_NOT_EXIST)
 
         return course_list
 
-    async def scrape_course_activities(self, course_code: str) -> list[dict]:
+    @handle_exception(error_type=ErrorType.SYSTEM_ERROR)
+    async def scrape_course(
+        self,
+        course_code: dict,
+        course: Optional[dict] = None,
+        raise_exception: bool = True,
```

Ruff (0.12.2)

Boolean-typed positional argument in function definition

(FBT001)